plugins/catalog-graph/src/components/CatalogGraphPage/SelectedOwnbyFilter.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { alertApiRef, useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import {
+  Box,
+  Checkbox,
+  FormControlLabel,
+  makeStyles,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+const useStyles = makeStyles({
+  formControl: {
+    maxWidth: 300,
+  },
+});
+
+export type Props = {
+  value: string[] | undefined;
+  onChange: (value: string[] | undefined) => void;
+};
+
+export const SelectedOwnbyFilter = ({ value, onChange }: Props) => {
+  const classes = useStyles();
+  const alertApi = useApi(alertApiRef);
+  const catalogApi = useApi(catalogApiRef);
+
+  const { error, value: ownedby } = useAsync(async () => {
+    return await catalogApi
+      .getEntityFacets({ facets: ['owner'] })
```

Comment:
Unfortunately you can't use the facets endpoint to get the list of possible users, since the shape of the relations are

```json
"relations": [
  {
    "type": "ownedBy",
    "sourceRef": "component:default/my-service",
    "targetRef": "group:default/my-team"
  }
]
```

Maybe you could just get all _potential_ owners instead? I.e. getting all entities, filtered to `kind` being `User` or `Group`. To reduce the response size of that, you could also just select `kind`, `metadata.namespace`, and `metadata.name` as the returned fields.

Reply:
Or maybe even better - if you could populate this list with what are all of the possible values _if all other filters are set the way they are_. As in, if the graph before your PR would show nodes that have two possible different owners (in their relations), then this select box should show only those two. And if the other filters change (ie if the "result set" when fetching all of that changes) then the candidates in this box change too.

Or will it be a use case to start from _only ownership_ and no other filters?

Reply:
@freben Agree with you, it's good to have that use case(select box should show only relevant owners) covered, at this moment I don't have any workaround, would appreciate your input on that.